packages/legend-query/src/components/QueryBuilderSetupPanel.tsx

```diff
@@ -50,8 +50,9 @@ export const QueryBuilderSetupPanel = observer(
       ? { value: querySetupState._class, label: querySetupState._class.name }
       : null;
     const changeClass = (val: PackageableElementOption<Class>): void => {
+      queryBuilderState.resetApp();
```

Comment:
This looks odd to me, by switching the order like this, we will miss the `refreshTreeData` call after setting the class, won't we?

Reply:
Logic will be moved to `changeClass` see overall comment